Microsoft.Toolkit.Uwp.UI.Controls/Graph/SharePointFiles/SharePointFiles.cs

```diff
@@ -0,0 +1,558 @@
+        {
+            if (string.IsNullOrEmpty(rawDocLibUrl))
+            {
+                return rawDocLibUrl;
+            }
+
+            rawDocLibUrl = WebUtility.UrlDecode(rawDocLibUrl);
+
+            Match match = Regex.Match(rawDocLibUrl, @"(https?://([^/]+)((/[^/?]+)*?)(/[^/?]+))(/(Forms/\w+.aspx)?)?(\?.*)?$", RegexOptions.IgnoreCase);
+            string docLibUrl = match.Groups[1].Value;
+            string hostName = match.Groups[2].Value;
+            string siteRelativePath = match.Groups[3].Value;
+            if (string.IsNullOrEmpty(siteRelativePath))
+            {
+                siteRelativePath = "/";
+            }
+
+            Site site = await _graphClient.Sites.GetByPath(siteRelativePath, hostName).Request().GetAsync();
+            ISiteDrivesCollectionPage drives = await _graphClient.Sites[site.Id].Drives.Request().GetAsync();
+
+            Drive drive = drives.SingleOrDefault(o => WebUtility.UrlDecode(o.WebUrl).Equals(docLibUrl, StringComparison.CurrentCultureIgnoreCase));
+            if (drive == null)
+            {
+                throw new Exception("Drive not found");
+            }
+
+            return _graphClient.Drives[drive.Id].RequestUrl;
+        }
+
+        private async Task InitDrive(string driveUrl)
+        {
+            try
+            {
+                HttpRequestMessage message = new HttpRequestMessage(HttpMethod.Get, driveUrl);
+                await _graphClient.AuthenticationProvider.AuthenticateRequestAsync(message);
+
+                HttpResponseMessage result = await _graphClient.HttpProvider.SendAsync(message);
+                if (result.StatusCode == HttpStatusCode.OK)
+                {
+                    string json = await result.Content.ReadAsStringAsync();
+                    Drive drive = JsonConvert.DeserializeObject<Drive>(json);
+                    if (drive != null)
+                    {
+                        _driveId = drive.Id;
+                        _driveItemPath.Clear();
+                        DriveItem rootDriveItem = await _graphClient.Drives[_driveId].Root.Request().GetAsync();
+                        _driveItemPath.Push(rootDriveItem.Id);
+                        await LoadFiles(rootDriveItem.Id);
+                        _back.Visibility = Visibility.Collapsed;
+                    }
+                }
+            }
+            catch
+            {
+            }
+        }
+
+        private async Task LoadFiles(string driveItemId, int pageIndex = 0)
+        {
+            IsDetailPaneVisible = false;
+            if (!string.IsNullOrEmpty(_driveId))
+            {
+                try
+                {
+                    _cancelLoadFile.Cancel(false);
+                    _cancelLoadFile.Dispose();
+                    _cancelLoadFile = new CancellationTokenSource();
+                    _list.Items.Clear();
+                    _download.Visibility = Visibility.Collapsed;
```

Comment:
You should expose a 'mode' type property and use that to toggle all your visibility based on which phase of the process you're in.

Reply:
done